### PR TITLE
[wasm][bcl] Fix `System.Diagnostics.Trace.TraceError` crash.

### PR DESCRIPTION
**Pull request by kjpou1**
```
Error: System.IndexOutOfRangeException: Index was outside the bounds of the array.
```

- WASM does not have access to the command line arguments through the browser so the length is 0 causing index out of range error.
- The zero index of the command line arguments is what decides the `AppName` so under WASM the AppName will be blanks.

fixes https://github.com/mono/mono/issues/17490

### CODE REVIEW

**Comment by kjpou1** [comment]
@monojenkins build failed

File: mcs/class/referencesource/System/compmod/system/diagnostics/TraceInternal.cs [hunk 1]
```diff
@@ -67,7 +67,13 @@ internal static class TraceInternal {
 #if MONO_FEATURE_CAS
                     new EnvironmentPermission(EnvironmentPermissionAccess.Read, "Path").Assert();
 #endif
+#if WASM
+                    // Wasm does not have access to command line arguments through browser
+                    if (Environment.GetCommandLineArgs().Length > 0)
+                        appName = Path.GetFileName(Environment.GetCommandLineArgs()[0]);
```
[the remark this answers is not in the document]
Comment (kjpou1) [reply mcs/class/referencesource/System/compmod/system/diagnostics/TraceInternal.cs]: Addressed in commit below.  Thanks zoltan

**Comment by vargaz** [comment]
Note that the #ifdef could be removed and the wasm version used everywhere.

**Comment by kjpou1** [comment]
review note addressed in recent commit.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**